### PR TITLE
Update detailed description of `constructAndSignVoteM-continue2.step₃-contract` proof due eliminating `VoteWithMeta`

### PR DESCRIPTION
**Pull request by mark-moir**
The description of this proof changed somewhat due to eliminating `VoteWithMeta`.

### CODE REVIEW

**Comment by cwjnkins** [comment]
I think you wanted to merge this into `chris-dijkstra-monad`, not `main`.

File: LibraBFT/Impl/Consensus/SafetyRules/Properties/SafetyRules.agda [hunk 1]
```diff
@@ -138,22 +141,22 @@ module ConstructAndSignVoteM where
     -- help ourselves understand.
 
     step₃-contract
-        : ∀ rm pre safetyData voteData ledgerInfo →
-          (RWST-weakestPre (step₃ safetyData voteData author ledgerInfo)
+        : ∀ rm pre safetyData1 voteData ledgerInfo →
```
Comment (cwjnkins) [inline LibraBFT/Impl/Consensus/SafetyRules/Properties/SafetyRules.agda:144]: I changed this to `safetyData1` to match the name used in the implementation model.

File: LibraBFT/Impl/Consensus/SafetyRules/Properties/SafetyRules.agda [hunk 2]
```diff
@@ -207,21 +208,23 @@ module ConstructAndSignVoteM where
 
       (RWST-bind
          (RWST-bind
-            RWST-get
-            (RWST-put "lSafetyData ∙= (safetyData1 & sdLastVote ?~ vote)"))   -- modifies the state returned by
-                                                                              -- RWST-get
-         (λ _ → RWST-return (inj₂ "Vote∙new vote mvsNew"))            -- The Unit returned by RWST-bind
-                                                                              -- via RWST-put is ignored
+            (RWST-gets id)                                                                -- Fetch the state.
```
Comment (cwjnkins) [inline LibraBFT/Impl/Consensus/SafetyRules/Properties/SafetyRules.agda:211]: `RWST-gets` is now the primitive operation, not `RWST-get`.

File: LibraBFT/Impl/Consensus/SafetyRules/Properties/SafetyRules.agda [hunk 3]
```diff
@@ -207,21 +208,23 @@ module ConstructAndSignVoteM where
...
    show how we instantiate the rules in the next step):
 
      RWST-weakestPre
       (RWST-bind
          (RWST-bind                                                              = m
-            RWST-get
-            (RWST-put "lSafetyData ∙= (safetyData1 [ sdLastVote ?= vote ])"))
-         (λ _ → RWST-return (inj₂ "Vote∙new vote mvsNew"))               = f
+            (RWST-gets id)
+            (λ st → RWST-put (st & lSafetyData ∙~ ("safetyData1" & sdLastVote ?~ "vote")))
```
Comment (cwjnkins) [inline LibraBFT/Impl/Consensus/SafetyRules/Properties/SafetyRules.agda:226]: I modified the shorthand somewhat to be more accurate (hopefully without too much impact on the readability). I think its important that, from reading this line, it is immediate that the expression is a function.

File: LibraBFT/Impl/Consensus/SafetyRules/Properties/SafetyRules.agda [hunk 4]
```diff
@@ -230,37 +233,37 @@ module ConstructAndSignVoteM where
...
        pre                                                                       = pre
 
-   Now applying the definition of RWST-weakestPre RWST-get, we want:
+   Now applying the definition of RWST-weakestPre RWST-gets, we want:
 
      (RWST-weakestPre-bindPost
-         unit                                                                    = ev
-         (RWST-put "lSafetyData ∙= (safetyData1 & sdLastVote ?~ vote)")          = f
-         (RWST-weakestPre-bindPost unit                                          = Post
-           (λ _ → RWST-return (inj₂ "Vote∙new vote mvsNew"))
```
Comment (cwjnkins) [inline LibraBFT/Impl/Consensus/SafetyRules/Properties/SafetyRules.agda:259]: This appeared to be a somewhat inconsistent use of the shorthand.

File: LibraBFT/Impl/Consensus/SafetyRules/Properties/SafetyRules.agda [hunk 5]
```diff
@@ -269,22 +272,26 @@ module ConstructAndSignVoteM where
 
      ∀ r → r ≡ pre →
        RWST-weakestPre
-         (RWST-put "lSafetyData ∙= (safetyData1 & sdLastVote ?~ vote)" pre)
+         (RWST-put (r & lSafetyData ∙~ ("safetyData1" & sdLastVote ?~ "vote")))
```
Comment (cwjnkins) [inline LibraBFT/Impl/Consensus/SafetyRules/Properties/SafetyRules.agda:275]: With the new shorthand, the substitution of `r` (which is a fairly important point, I think) into the function body becomes clear.

**Review by cwjnkins — APPROVED** [review]
Looks good to me. I'm leaving this open only so we can further discuss the changes I've pushed, if you would like to.

**Comment by mark-moir** [comment]
@cwjnkins Thanks for the careful proofreading and excellent improvements to my "essay".  There was still one spelling mistake (of mine), so I thought I would get the last word on this one.  But I see you merged, so you got the last word. 
 Well played :slightly_smiling_face:.

**Pull request merged**